test/Microsoft.Azure.ServiceBus.UnitTests/TokenProviderTests.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.ServiceBus.UnitTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.ServiceBus.Primitives;
+    using Microsoft.IdentityModel.Clients.ActiveDirectory;
+    using Xunit;
+
+    public class TokenProviderTests : SenderReceiverClientTestBase
+    {
+        /// <summary>
+        /// This test is for manual only purpose. Fill in the tenant-id, app-id and app-secret before running.
+        /// </summary>
+        /// <returns></returns>
+        [Fact]
+        [DisplayTestMethodName]
+        async Task UseITokenProviderWithAad()
+        {
+            var tenantId = "";
+            var aadAppId = "";
+            var aadAppSecret = "";
+
+            if (string.IsNullOrEmpty(tenantId))
```

Comment:
It's ok to leave it for later (with an accompanying issue), though @serkantkaraca please consider https://github.com/Azure/azure-service-bus-dotnet/pull/383#discussion_r156526267.

Reply:
Thx Sean. We will increase test coverage going forward. These changes will be release as preview tomorrow. So we are in a little hurry right now and relying on manual tests only.

---
In reply to: [156526756](https://github.com/Azure/azure-service-bus-dotnet/pull/385#discussion_r156526756) [](ancestors = 156526756)